modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/utils/imagePropsTransformer.ts

```diff
@@ -3,21 +3,82 @@
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
 
-export default function imagePropsTransformer(
-	imageData: any | string
-): React.ImgHTMLAttributes<HTMLImageElement> | string | undefined {
-	if (typeof imageData === 'string') {
-		return imageData;
-	}
+import getValueFromItem from '../../utils/getValueFromItem';
+
+interface IDocsAndMediaImageProps {
+	id: number;
+	link: {
+		href: string;
+		label: string;
+	};
+	name: string;
+}
 
-	if (typeof imageData === 'object') {
+interface IImageProps {
+	itemData?: any;
+	name?: string;
+	options?: {
+		label?: string;
+		labelKey?: string;
+		shape?: 'circle' | 'rounded' | 'user-icon';
+		size?: 'lg' | 'sm' | 'xl';
+	};
+	value?:
+		| string
+		| IDocsAndMediaImageProps
+		| React.ImgHTMLAttributes<HTMLImageElement>;
+}
 
-		// Docs & media object
+export default function imagePropsTransformer(
+	imageData: IDocsAndMediaImageProps | IImageProps | string | undefined
+): React.ImgHTMLAttributes<HTMLImageElement> | undefined {
+	let imageProps: React.ImgHTMLAttributes<HTMLImageElement> = {
+		alt: '',
+		src: undefined,
+	};
 
-		return {
-			alt: imageData?.link.label,
-			id: imageData?.id,
-			src: imageData?.link.href,
+	if (!imageData || !Object.keys(imageData).length) {
+		return undefined;
+	}
+
+	if (typeof imageData === 'string') {
+		imageProps = {...imageProps, src: imageData};
+	}
+	else if ('value' in imageData && typeof imageData.value === 'string') {
+		imageProps = {
+			...imageProps,
+			alt:
+				imageData.options?.label ||
```

Comment:
Cannot find any usage of this `options` but keep it there anyway. Remove the default `Thumbnail` value for the `alt` attribute following a11y guidelines.
The problem with the `options.label` is that it could return a non expected value when it appears (cell renderers for the table view return renderer option with a `label` property).

Reply:
Yeah, it's nearly impossible to tell all existing formats of returned data. This was almost certainly some usage in commerce, who knows if it still exists. We can keep this code for now, deal with the potential issue if it ever comes up. At some point commerce should migrate to Objects, at that point we'll be able to cleanup a lot of "special cases" from FDS.